modules/dnn/test/test_layers.cpp

```diff
@@ -356,8 +362,14 @@ TEST_P(Test_Caffe_layers, Conv_Elu)
     net.setPreferableBackend(backend);
     net.setPreferableTarget(target);
     Mat out = net.forward();
-
-    normAssert(ref, out, "", default_l1, default_lInf);
+    
+    double l1 = default_l1, lInf = default_lInf;
+    if (target == DNN_TARGET_CUDA_FP16)
+    {
+        l1 = 2;
+        lInf = 3;
```

Comment:
Also seems too big. Please specify the range of reference values.

Reply:
```
[ RUN      ] Test_Caffe_layers.Conv_Elu/1, where GetParam() = CUDA/CUDA_FP16
[CUDA] Convolution elu/Conv2D
[CUDA] ELU elu
Ref. Range: -0.255777 0.0283771
[       OK ] Test_Caffe_layers.Conv_Elu/1 (26 ms)
```

I  think it's just ELU being very inaccurate. The FP32 target uses a numerically stable `expm1` for the activation while the FP16 uses `exp(x) - 1`. I can try to write a numerically stable version for the half precision kernel but that will be considerably slower and would suffer from wrap divergence.

Reply:
Bug fixed in #16045.

New thresholds:
```
    double l1 = 1e-5, lInf = 1e-4;
    if (target == DNN_TARGET_CUDA_FP16)
    {
        l1 = 0.0002;
        lInf = 0.0005;
    }
```